server/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -205,7 +209,18 @@ private void submitUnbatchedTask(@SuppressWarnings("SameParameterValue") String
         clusterService.submitUnbatchedStateUpdateTask(source, task);
     }
 
-    private void executeHealth(
+    private void forkAndExecuteHealth(
+        final CancellableTask task,
+        final ClusterHealthRequest request,
+        final ClusterState currentState,
+        final ActionListener<ClusterHealthResponse> listener,
+        final int waitCount,
+        final Consumer<ClusterState> onNewClusterStateAfterDelay
+    ) {
+        executor.execute(() -> executeHealthOnCurrentThread(task, request, currentState, listener, waitCount, onNewClusterStateAfterDelay));
```

Comment:
nit I am not sure I like creating a function for a one-liner. I might have preferred simply re-using this line in the places you use the function.